borsh-derive/src/internals/deserialize/mod.rs

```diff
@@ -61,41 +67,63 @@ fn process_field(
         if needs_bounds_derive {
             generics.deserialize_visitor.visit_field(field);
         }
-        field_output(field_name, cratename, parsed.deserialize_with)
+        field_output::<IS_ASYNC>(
+            field_name,
+            &field.ty,
+            cratename,
+            if IS_ASYNC {
+                cfg_if! {
+                    if #[cfg(feature = "async")] {
+                        parsed.deserialize_with_async
+                    } else {
+                        None
+                    }
+                }
+            } else {
+                parsed.deserialize_with
+            },
```

Comment:
this looks like it can be extracted to a method `get_deser_with`, similar to `internals::attributes::field::Attributes::get_bounds`